### DISCUSSION TITLE
Upload files in chunks

### DISCUSSION THREAD

**Discussion by joselcvarela** [directus/directus#22311] [Feature Requests]
### Summary

Currently, the way of uploading files is all or nothing.
In other words, we need to keep connection active otherwise upload will fail.
The same happens for any VPNs, CDNs or any other servers in the middle.

Also, some servers in the middle have a maximum timeout to receive a response from origin.
While doing uploads for big files, this timeout will be reached as the upload may take more than 5 minutes to accomplish

### Motivation

We pretend to have this in order to be able to upload big files for use cases like video streaming, digital signage, games content, etc.

### Detailed Design

I am not sure what kind of implementation we should follow, but in [another issue](https://github.com/directus/directus/issues/21852) it was mentioned the [tus](https://tus.io/) protocol.

So we might implement a solution based in this protocol.

### Must Have:
- Upload files in chunks

### Should Have:
- Upload progress bar

### Could Have:
- Resumable uploads

### Won't Have:
- N/A

### Drawbacks

This is a whole new feature that needs to be made that includes App and Api, so it will take more than 1 month to be made.

### Alternatives

The current alternative is to inject the asset directly into the storage and create the respective entry on `directus_files`

### Adoption Strategy

No breaking changes are expected

### Unresolved Questions

What protocol will be used? [tus](https://tus.io/)? [resumable.js](https://github.com/23/resumable.js)? other?

**Comment by rijkvanzanten** [comment]
Supporting TUS feels like the way to go here 👍🏻

**Comment by ComfortablyCoding** [comment]
Closing as we now support chunking and resumable uploads via TUS for all supported drivers
